### PR TITLE
fix(connectors): fixes MGDCTRS-753, use optimistic locking in connector updates

### PR DESCRIPTION
**Pull request by dhirajsb**
## Description
Added optimistic locking using connector version in where condition to ensure the connector didn't change just before the update. 

## Verification Steps
CI tests should pass. 

## Checklist (Definition of Done)
- [X] All acceptance criteria specified in JIRA have been completed
~~- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)~~
~~- [ ] Documentation added for the feature~~
- [x] CI and all relevant tests are passing
- [ ] Code Review completed
~~- [ ] Verified independently by reviewer~~
~~- [ ] Required metrics/dashboards/alerts have been added (or PR created).~~
~~- [ ] Required Standard Operating Procedure (SOP) is added.~~
~~- [ ] JIRA has created for changes required on the client side~~

### CODE REVIEW

File: internal/connector/internal/services/connectors.go [hunk 1]
```diff
@@ -286,19 +286,18 @@ func selectConnectorWithConditions(dbConn *gorm.DB) *gorm.DB {
...
 	}
 
 	dbConn := k.connectionFactory.New()
-	if err := dbConn.Model(resource).Updates(resource).Error; err != nil {
+	update := dbConn.Model(resource).Where("id = ? AND version = ?", resource.ID, resource.Version).Updates(resource)
+	if err := update.Error; err != nil {
 		return errors.GeneralError("failed to update: %s", err.Error())
 	}
+	if update.RowsAffected == 0 {
+		return errors.BadRequest("resource version changed")
```
Comment (lburgazzoli) [inline internal/connector/internal/services/connectors.go:299]: should this be 409 (Conflict) ?

**Review by valdar — CHANGES_REQUESTED** [review]
Can we add a basic test with 2 request with the same version and check that the second one fails?

File: internal/connector/internal/services/connectors.go (hunk 1 above)
Comment (dhirajsb) [reply internal/connector/internal/services/connectors.go]: agreed, changed to conflict.

**Comment by valdar** [comment]
> Can we add a basic test with 2 request with the same version and check that the second one fails?

we can not in the easy way I described, unfortunately.

**Review by valdar — APPROVED** [review]

**Pull request merged**